contracts/AsyncFHE.sol

```diff
@@ -0,0 +1,4751 @@
+// This file is auto-generated by solgen.
+// SPDX-License-Identifier: BSD-3-Clause-Clear
+// solhint-disable one-contract-per-file
+
+pragma solidity >=0.8.19 <0.9.0;
+
+// import {Console} from "@fhenixprotocol/contracts/utils/debug/Console.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+import {Precompiles, FheOps} from "./FheOS.sol";
+
+type ebool is uint256;
+type euint8 is uint256;
+type euint16 is uint256;
+type euint32 is uint256;
+type euint64 is uint256;
+type euint128 is uint256;
+type euint256 is uint256;
+type eaddress is uint256;
+
+struct inEbool {
+    bytes data;
+    int32 securityZone;
+}
+struct inEuint8 {
+    bytes data;
+    int32 securityZone;
+}
+struct inEuint16 {
+    bytes data;
+    int32 securityZone;
+}
+struct inEuint32 {
+    bytes data;
+    int32 securityZone;
+}
+struct inEuint64 {
+    bytes data;
+    int32 securityZone;
+}
+struct inEuint128 {
+    bytes data;
+    int32 securityZone;
+}
+struct inEuint256 {
+    bytes data;
+    int32 securityZone;
+}
+struct inEaddress {
+    bytes data;
+    int32 securityZone;
+}
+
+struct SealedArray {
+  bytes[] data;
+}
+
+/// @dev Utility structure providing clients with type context of a sealed output string.
+/// Return type of `FHE.sealoutputTyped` and `sealTyped` within the binding libraries.
+/// `utype` representing Bool is 13. See `FHE.sol` for more.
+struct SealedBool {
+    string data;
+    uint8 utype;
+}
+
+/// @dev Utility structure providing clients with type context of a sealed output string.
+/// Return type of `FHE.sealoutputTyped` and `sealTyped` within the binding libraries.
+/// `utype` representing Uints is 0-5. See `FHE.sol` for more.
+/// `utype` map: {uint8: 0} {uint16: 1} {uint32: 2} {uint64: 3} {uint128: 4} {uint256: 5}.
+struct SealedUint {
+    string data;
+    uint8 utype;
+}
+
+/// @dev Utility structure providing clients with type context of a sealed output string.
+/// Return type of `FHE.sealoutputTyped` and `sealTyped` within the binding libraries.
+/// `utype` representing Address is 12. See `FHE.sol` for more.
+struct SealedAddress {
+    string data;
+    uint8 utype;
+}
+
+// ================================
+// \/ \/ \/ \/ \/ \/ \/ \/ \/ \/ \/
+// TODO : CHANGE ME AFTER DEPLOYING
+// /\ /\ /\ /\ /\ /\ /\ /\ /\ /\ /\
+// ================================
+//solhint-disable const-name-snakecase
+address constant TASK_MANAGER_ADDRESS = address(129);
```

Comment:
@Cashmaney @liorbond @toml01  I'm wondering, do we wanna have auto deployments through CI ? 
`AsyncFHE.sol` is coupled to a specific already-deployed TaskManager right now, to prevent sending it's address on each operation, kinda like before with the precompiles, but without the ability to have predefined constant address.. 
Given that it's gonna be deployed on multiple chains, we either need to update it manually here upon deployment, create folders per chain, or automate it (which I'm not sure worth the effort right now)

Reply:
I wouldn't deploy now, as soon as we will have the networks we are working with, we can have it published in a good structutre such as "fhenixprotocol/contracts/ethereum/FHE.sol"

Reply:
You should also check out the `CREATE2` opcode, which enables you to create a contract in a deterministic predefined address

Reply:
COOL!! @toml01 I can also use the salt inside the calculated address for versioning purposes 🤯

Reply:
Hmm you mean a different address for every version? I think we'd rather have an upgradable contract living always at the same address, no?

Reply:
hmmm actually makes sense, it doesn't reduce confidence and adoption to have such kind of "backdoors" for ourselves?